velox/exec/tests/utils/QueryAssertions.h

```diff
@@ -98,6 +101,65 @@ class TestScopedAbortInjection {
   ~TestScopedAbortInjection();
 };
 
+class ScopedOOMInjector {
+ public:
+  ScopedOOMInjector(
+      const std::function<bool()>& oomCondition,
+      uint64_t oomCheckIntervalMs)
+      : oomCondition_(oomCondition), oomCheckIntervalMs_(oomCheckIntervalMs) {}
+
+  void enable() {
+    // Since this relies on TestValue to trigger the OOMs, it only supports
+    // debug builds.
+#ifdef NDEBUG
+    VELOX_FAIL("OOM injection can only be used in debug builds");
+#endif
+
+    VELOX_CHECK(!enabled_, "Already enabled");
+    enabled_ = true;
+
+    // Make sure TestValues are enabled.
+    common::testutil::TestValue::enable();
+
+    common::testutil::TestValue::set(
+        kInjectionPoint,
+        std::function<void(memory::MemoryPool*)>([&](memory::MemoryPool*) {
+          auto currentTime = now();
+          if (currentTime - lastOomCheckTime_ >= oomCheckIntervalMs_) {
+            lastOomCheckTime_ = currentTime;
+            if (oomCondition_()) {
+              LOG(INFO) << "<-- Triggering OOM --";
+              VELOX_MEM_POOL_CAP_EXCEEDED("Inject OOM");
+            }
+          }
+        }));
+  }
+
+  ~ScopedOOMInjector() {
+    common::testutil::TestValue::clear(kInjectionPoint);
+    enabled_ = false;
+  }
+
+ private:
+  static const std::string kInjectionPoint;
+  // If more than one instance of this class is enabled, they'll overwrite
+  // each other, so make sure no one does this by accident.
+  static bool enabled_;
```

Comment:
Can you just make this std::atomic since this is only used in debug build? Thanks!